### DISCUSSION TITLE
Virtual thread support

### DISCUSSION THREAD

**Discussion by StevenCurran** [jOOQ/jOOQ#14901] [Q&A] [answered]
Hey all, question on virtual thread support. 

Is this mainly dependent on the driver supporting it? Eg pgjdbc has replaced relevant iO sync calls with j.I.c.locks. 

Does this mean jooq can just be called from a virtual thread safely or is there some changes still required in the library? 

Was not able to find anything on issues / discussions here, so apologies if I missed it.

Thanks.

**Comment by lukaseder** [comment, marked as answer]
jOOQ doesn't create any threads, nor does it block on any operations itself other than via JDBC, so I don't think there's anything jOOQ specific to this discussion. jOOQ won't even notice if JDBC drivers / connection pools switch their concurrency model.

I hope this helps.

**Reply by StevenCurran** [reply]
Yes it does, thank you Lukas.
